addons/crm_mail_plugin/controllers/mail_plugin.py

```diff
@@ -53,14 +53,29 @@ def _fetch_partner_leads(self, partner, limit=5, offset=0):
 
     def _get_contact_data(self, partner):
         contact_values = super(MailPluginController, self)._get_contact_data(partner)
+
+        if not request.env.user.has_group('sales_team.group_sale_salesman'):
+            return contact_values
+
         if not partner:
             contact_values['leads'] = []
         else:
             contact_values['leads'] = self._fetch_partner_leads(partner)
         return contact_values
 
     def _mail_content_logging_models_whitelist(self):
-        return super(MailPluginController, self)._mail_content_logging_models_whitelist() + ['crm.lead']
+        models_whitelist = super(MailPluginController, self)._mail_content_logging_models_whitelist()
+        if not request.env.user.has_group('sales_team.group_sale_salesman'):
```

Comment:
Wouldn't be a lot cleaner to do something like:
```py
if not request.env['crm.lead'].check_access_rights('read', raise_exception=False):
    return models_whitelist
```

That way it matches the check done in mail_thread?
(We would need to check access rules too but that's record-dependent so it can't be done here).

Currently you check for the salesman group, but what if other groups also have the rights to create and manage leads, based on a customization for example? It it necessary to restrict to salesman?

(Maybe that applies to other methods too? To double-check).

Reply:
yes, good idea :) 

Instead of `_user_can_create_contact` I can do the same as well